code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -767,27 +774,48 @@
 
 // This proc attempts to find a new ruler to lead the hive.
 /datum/hive_status/proc/update_ruler()
-	if(living_xeno_ruler)
+	if(isxenoqueen(living_xeno_ruler))
 		return //No succession required.
 
 	var/mob/living/carbon/xenomorph/successor
+	var/list/xenos = get_all_xenos(TRUE)
+	var/list/candidates = list()
+	var/index = 1
+	var/secondind = 1
+
+	for(var/mob/living/carbon/xenomorph/x in xenos)
+		if(x.xeno_caste.can_flags & CASTE_CAN_BE_RULER) //We basically wanna pool all castes who are eligible
+			candidates.Insert(secondind, xenos[index])
+			secondind++
+		index++
+	index = 1
+
+	for(var/mob/living/carbon/xenomorph/potential_successor in candidates)
+		if(isxenoqueen(potential_successor) || isxenoshrike(potential_successor) || isxenoking(potential_successor)) // prio to Queen / Shrike / King
+			successor = candidates[index]
+			if(living_xeno_ruler) // T4 should take over for the T3 and we want to remove their abilities
+				living_xeno_ruler.remove_ruler_abilities()
+			if(isxenoqueen(successor)) // Queen rules over all
+				break
+		index++
 
-	// TO DO: this shouldn't be a strict type check and should account for strains
-	var/list/candidates = xenos_by_typepath[/datum/xeno_caste/queen] + xenos_by_typepath[/datum/xeno_caste/shrike] + xenos_by_typepath[/datum/xeno_caste/king] + xenos_by_typepath[/datum/xeno_caste/dragon] + xenos_by_typepath[/datum/xeno_caste/king/conqueror]
-	if(length(candidates)) //Priority to the queens.
-		successor = candidates[1] //First come, first serve.
+	if(successor == null && !living_xeno_ruler)
+		successor = candidates[1] // Basically set whoever is at the top if there is no Queen / Shrike  / King
+	else if(successor == null && living_xeno_ruler) // We dont want other T3s evolving to just takeover
+		return
```

Comment:
THis seems way overengineered? why not just get xenos by type followed by get xenos from tier, and using += instead of insert

Reply:
Got errors when adding to the list via += for some reason. I'll re-try again